apps/utils/locust_tests/common/data.py

```diff
@@ -3,16 +3,34 @@
 import datetime
 import logging
 from argparse import Namespace
-from typing import Callable, List
+from typing import Any, Optional, Protocol, Union, runtime_checkable
+
+from common.types import CopyableEnumerable
+
+
+@runtime_checkable
+class NonSamplingLoadFunction(Protocol):
+    def __call__(self, uri: str) -> CopyableEnumerable[Any]:
+        ...
+
+
+@runtime_checkable
+class SamplingLoadFunction(Protocol):
+    def __call__(
+        self, uri: str, table_sample_pct: Optional[float] = None
+    ) -> CopyableEnumerable[Any]:
+        ...
```

Comment:
These [`Protocol`]() classes are ["Callback Protocols"](https://mypy.readthedocs.io/en/latest/protocols.html#callback-protocols) and serve as _much_ stronger type guarantees for the type of the function that's provided as the data loading function. These `Protocol`s are used to ensure only the following functions pass static type analysis when passed into `load_from_parsed_opts()`/`load_from_uri()`:

```
def load_function(uri: str) -> CopyableEnumerable[Any]: ...
def load_function(uri: str, table_sample_pct: Optional[float] = None) -> CopyableEnumerable[Any] ...
```

Anything else will be rejected by the type checker.

Additionally, they are both `@runtime_checkable`, as it is necessary to check _which_ type of `LoadFunction` the given `Callable` is during runtime. Without this attribute, the below `LoadFunction` `Union` doesn't work.